### PR TITLE
Feature/jr2 524 index jobs by x28 avam and sbn codes

### PR DESCRIPTION
**Pull request by birom**
I want to merge this together with the related chages in the front end and the integration projects.

### CODE REVIEW

File: src/main/java/ch/admin/seco/service/job/domain/Occupation.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        Occupation that = (Occupation) o;
+
+        if (id != null ? !id.equals(that.id) : that.id != null) {
```
Comment (varusakov) [inline src/main/java/ch/admin/seco/service/job/domain/Occupation.java]: `Occupation` is entity. I think comparison by id is enough in this case.

File: src/main/java/ch/admin/seco/service/job/domain/valueobject/OccupationCode.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+        if (value != that.value) {
+            return false;
+        }
+        return type == that.type;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = value;
```
Comment (varusakov) [inline src/main/java/ch/admin/seco/service/job/domain/valueobject/OccupationCode.java]: Same as above

File: src/main/java/ch/admin/seco/service/job/domain/Occupation.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+            return false;
+        }
+        if (experience != that.experience) {
```
Comment (varusakov) [inline src/main/java/ch/admin/seco/service/job/domain/Occupation.java]: Could you use `Objects.equals` as we do in another places?

File: src/main/java/ch/admin/seco/service/job/domain/Occupation.java (hunk 1 above)
Comment (michael-wirth) [reply src/main/java/ch/admin/seco/service/job/domain/Occupation.java]: The methods `equals`, `hashcode` and `toString` shoudl be generated by Intellij

File: src/main/java/ch/admin/seco/service/job/domain/valueobject/Occupation.java [hunk 4]
```diff
@@ -1,106 +0,0 @@
...
-import javax.validation.constraints.Max;
-import javax.validation.constraints.Min;
-import javax.validation.constraints.NotNull;
-
-import ch.admin.seco.service.job.domain.enumeration.Experience;
-
-/**
- * A Occupation.
- */
-@Embeddable
```
Comment (michael-wirth) [inline src/main/java/ch/admin/seco/service/job/domain/valueobject/Occupation.java]: Why was this changed from an `Embeddable` into an `Entity`? Was there technical issue or are nested `@ElementCollection`s not allowed?

Comment (birom) [reply src/main/java/ch/admin/seco/service/job/domain/valueobject/Occupation.java]: Nested @ElementCollection s are not allowed.

File: src/main/java/ch/admin/seco/service/job/domain/Occupation.java (hunk 1 above)
Comment (birom) [reply src/main/java/ch/admin/seco/service/job/domain/Occupation.java]: Thank you, I fixed it.
